pydantic/main.py

```diff
@@ -887,7 +887,9 @@ def create_model(
         `foobar=(str, ...)` or `foobar=123`, or, for complex use-cases, in the format
         `<name>=<FieldInfo>`, e.g. `foo=Field(default_factory=datetime.utcnow, alias='bar')`
     """
-    if __base__:
+    __base__ = cast(Type['Model'], __base__)  # to make mypy happy
+
+    if __base__ is not None:
```

Comment:
I don't understand why you need this. I thought only setting `'Model'` instead of `BaseModel` in the signature would be enough

Reply:
Without this cast `mypy` will fail with errors:
```
pydantic/main.py:896: error: Incompatible types in assignment (expression has type "Type[BaseModel]", variable has type "Optional[Type[Model]]")  [assignment]
pydantic/main.py:927: error: Argument 2 to "type" has incompatible type "Tuple[Optional[Type[Model]]]"; expected "Tuple[type, ...]"  [arg-type]
```

As a solution for this problem, I decided to cast `__base__` to `Type['Model']` because it always will be of this type.

Reply:
Can you please give a working example ? I just tried to run your test without the cast and it works as expected. I'm probably missing something

Reply:
When I tried to run `mypy` stage without `cast` it fails. Logs of `mypy`:
```
$ make mypy
mypy pydantic
pydantic/main.py:896: error: Incompatible types in assignment (expression has type "Type[BaseModel]", variable has type "Optional[Type[Model]]")  [assignment]
pydantic/main.py:927: error: Argument 2 to "type" has incompatible type "Tuple[Optional[Type[Model]]]"; expected "Tuple[type, ...]"  [arg-type]
Found 2 errors in 1 file (checked 23 source files)
make: *** [mypy] Error 1
```

Reply:
Ah right! Then I would simply write
```python
    if __base__ is not None:
        if __config__ is not None:
            raise ConfigError('to avoid confusion __config__ and __base__ cannot be used together')
    else:
        __base__ = cast(Type['Model'], BaseModel)
```
as the cast is important only when we set `__bases__`. WDYT ?

Reply:
Thanks, you are totally right.